core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryPolicy.java

```diff
@@ -113,6 +113,10 @@ public static Builder builder(RetryMode retryMode) {
         Validate.paramNotNull(retryMode, "The retry mode cannot be set as null. If you don't want to set the retry mode,"
                                     + " please use the other builder method without setting retry mode, and the default retry"
                                     + " mode will be used.");
+        if (retryMode == RetryMode.ADAPTIVE_V2) {
+            throw new UnsupportedOperationException("ADAPTIVE_V2 is not supported by retry policies, use a RetryStrategy "
+                                                    + "instead");
+        }
```

Comment:
Can we move this check to `Builder` instead?

Reply:
Yes, we can, but any reason for that? I don't mind either way but seems to me that we don't validate anything for builders as those can be in a non-consistent state up until they are built. Let me know.

Reply:
I guess you can argue that it will create a better stack trace that points  to where the problem is instead of on a separated place, so makes sense. I will move it there.